### PR TITLE
80 update pytests

### PR DESCRIPTION
**Pull request by lisa-sousa**
Updated all tests, except the oligoset generation tests. Added missing tests for new functionalities. Removed all doc strings for now since most were outdated. Will be added again documentation issue.

### CODE REVIEW

File: oligo_designer_toolsuite/database/_oligo_database.py [hunk 1]
```diff
@@ -721,58 +575,140 @@ def get_sequence_oligoid_mapping(
...
-                oligo_attributes.update(new_oligo_attribute[oligo_id])
+                if oligo_id in new_oligo_attribute:
+                    oligo_attributes.update(format_oligo_attributes(new_oligo_attribute[oligo_id]))
+
+    def filter_database_by_region(self, remove_region: bool, region_ids: Union[str, List[str]]) -> None:
+
+        # Check formatting
+        region_ids = check_if_list(region_ids)
+        if self.database:
+            for region_id in self.database.keys():
```
Comment (francesco-campi) [inline oligo_designer_toolsuite/database/_oligo_database.py:637]: The code could be parallelized region-wise.

File: oligo_designer_toolsuite/database/_oligo_database.py [hunk 2]
```diff
@@ -721,58 +575,140 @@ def get_sequence_oligoid_mapping(
...
+            raise ValueError(
+                "Can not filter. Database is empty! Call the method load_database() or load_database_from_fasta() first."
+            )
+
+    def filter_database_by_oligo(self, remove_region: bool, oligo_ids: Union[str, List[str]]) -> None:
+
+        # Check formatting
+        oligo_ids = check_if_list(oligo_ids)
+        if self.database:
+            for region_id in self.database.keys():
```
Comment (francesco-campi) [inline oligo_designer_toolsuite/database/_oligo_database.py:652]: The code could be parallelized region-wise.

File: oligo_designer_toolsuite/database/_oligo_database.py [hunk 3]
```diff
@@ -721,58 +575,140 @@ def get_sequence_oligoid_mapping(
...
+        else:
+            raise ValueError(
+                "Can not filter. Database is empty! Call the method load_database() or load_database_from_fasta() first."
+            )
+
+    def filter_database_by_attribute_threshold(
+        self, attribute_name: str, attribute_thr: float, remove_if_smaller_threshold: bool
+    ) -> None:
+        oligos_to_delete = []
+        for region_id in self.database.keys():
```
Comment (francesco-campi) [inline oligo_designer_toolsuite/database/_oligo_database.py:668]: The code could be parallelized region-wise.

File: oligo_designer_toolsuite/database/_oligo_database.py [hunk 4]
```diff
@@ -721,58 +575,140 @@ def get_sequence_oligoid_mapping(
...
+
+    def filter_database_by_attribute_category(
+        self, attribute_name: str, attribute_category: Union[str, List[str]], remove_if_equals_category: bool
+    ) -> None:
+
+        # Check formatting
+        attribute_category = check_if_list(attribute_category)
+        oligos_to_delete = []
+
+        for region_id in self.database.keys():
```
Comment (francesco-campi) [inline oligo_designer_toolsuite/database/_oligo_database.py:694]: The code could be parallelized region-wise.

File: oligo_designer_toolsuite/database/_oligo_database_attributes.py [hunk 5]
```diff
@@ -3,216 +3,125 @@
...
 from oligo_designer_toolsuite.database import OligoDatabase
-from oligo_designer_toolsuite.utils import check_if_key_exists, check_if_list
+from oligo_designer_toolsuite.utils import check_if_list, flatten_attribute_list
 
 ############################################
 # Attrubite Calculation Class
 ############################################
 
 
 class OligoAttributes:
```
Comment (francesco-campi) [inline oligo_designer_toolsuite/database/_oligo_database_attributes.py:21]: All the computations of the attribues could be parallelized region-wise to speed up the computations.

**Review by francesco-campi — COMMENTED** [review]
Looks good to me. I just left a couple of very minor comments regarding possible optimizations of the code. However, I am not sure wheter the improvement would be noticable or not in terms of compute time. So, in case you thing the chages are not necessary, I ma prefectly fine with merging the code the way it is right now.

**Review by francesco-campi — APPROVED** [review]

File: oligo_designer_toolsuite/database/_oligo_database_attributes.py (hunk 5 above)
Comment (lisa-sousa) [reply oligo_designer_toolsuite/database/_oligo_database_attributes.py:21]: True, we will address it in the oligo attribute issue

File: oligo_designer_toolsuite/database/_oligo_database.py (hunk 1 above)
Comment (lisa-sousa) [reply oligo_designer_toolsuite/database/_oligo_database.py:637]: True, we will address it in the oligo attribute issue

File: oligo_designer_toolsuite/database/_oligo_database.py (hunk 3 above)
Comment (lisa-sousa) [reply oligo_designer_toolsuite/database/_oligo_database.py:668]: True, we will address it in the oligo attribute issue

File: oligo_designer_toolsuite/database/_oligo_database.py (hunk 2 above)
Comment (lisa-sousa) [reply oligo_designer_toolsuite/database/_oligo_database.py:652]: True, we will address it in the oligo attribute issue

File: oligo_designer_toolsuite/database/_oligo_database.py (hunk 4 above)
Comment (lisa-sousa) [reply oligo_designer_toolsuite/database/_oligo_database.py:694]: True, we will address it in the oligo attribute issue

**Pull request merged**